org-code-javabuilder/protocol/src/main/java/org/code/protocol/JavabuilderRuntimeException.java

```diff
@@ -19,6 +23,20 @@ protected JavabuilderRuntimeException(Enum key, Throwable cause) {
     this.key = key;
   }
 
+  protected JavabuilderRuntimeException(Enum key, Boolean logOnCreate) {
+    this(key);
+    if (logOnCreate) {
```

Comment:
Thanks for the context.  One advantage of logging higher up is that you can log exceptions that are created by the runtime or libraries but perhaps that as much of a concern here?

Having logging code in the exception itself feels weird to me.  Typically I think exceptions are logged where they are handled or up at the top-level.  But I don't feel that strongly about it if the logging isn't driving user behavior.

Reply:
I did a little bit of searching and maybe anti-pattern is a bit too strong but there does seem to be some agreement that it's not ideal.  Here's a thread I found on stackexchange: https://softwareengineering.stackexchange.com/questions/306032/should-i-log-errors-on-exception-throwing-constructors that has some viewpoints from others.

Reply:
Ok. After a bit more thought and looking through the thread you shared, I'm going to go ahead and move the logging to CodeBuilderWrapper. I couldn't find a strong reason for doing it one way or another, but one thing I've definitely learned through the years is that trusting people's instincts tends to pay off.

Reply:
A point in favor of using CodeBuilderWrapper for this: Thinking about it like Rails, we have one generic 500 page in our dashboard server that gets shown, regardless of the error type, and I'm guessing that's configured globally. Similarly, we'd want a globally configured way of handling these types of errors. This is a much more lightweight system (and this is the logger, rather than the 500 page), but that still seems like a design pattern that is helpful.